java/impl/aws/src/main/java/co/worklytics/psoxy/WebhookCollectionModeHandler.java

```diff
@@ -0,0 +1,174 @@
+package co.worklytics.psoxy;
+
+import co.worklytics.psoxy.aws.AwsContainer;
+import co.worklytics.psoxy.aws.SQSOutput;
+import co.worklytics.psoxy.aws.request.APIGatewayV2HTTPEventRequestAdapter;
+import co.worklytics.psoxy.gateway.HttpEventResponse;
+import co.worklytics.psoxy.gateway.ProcessedContent;
+import co.worklytics.psoxy.gateway.impl.BatchMergeHandler;
+import co.worklytics.psoxy.gateway.impl.InboundWebhookHandler;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2HTTPEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2HTTPResponse;
+import com.amazonaws.services.lambda.runtime.events.SQSEvent;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+import co.worklytics.psoxy.aws.DaggerAwsContainer;
+
+/**
+ * AWS lambda entrypoint that can handle BOTH:
+ *    - webhook requests from API Gateway / Function URL invocations, AND
+ *    - SQS events (presumed to be batches of webhooks)
+ */
+public class WebhookCollectionModeHandler implements RequestStreamHandler {
+
+    /**
+     * Static initialization allows reuse in containers
+     * {@link "https://aws.amazon.com/premiumsupport/knowledge-center/lambda-improve-java-function-performance/"}
+     */
+    static AwsContainer awsContainer;
+
+    static BatchMergeHandler batchMergeHandler;
+
+    static InboundWebhookHandler inboundWebhookHandler;
+
+    static ObjectMapper mapper;
+
+    static ObjectMapper sqsPayloadMapper;
+
+    static {
+        staticInit();
+    }
+
+    private static void staticInit() {
+        awsContainer = DaggerAwsContainer.create();
+        batchMergeHandler = awsContainer.batchMergeHandler();
+        inboundWebhookHandler = awsContainer.inboundWebhookHandler();
+        mapper = awsContainer.objectMapper();
+        sqsPayloadMapper = new ObjectMapper();
+        sqsPayloadMapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
+    }
+
+
+    @Override
+    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
+        // Read the full input stream into a tree
+        JsonNode rootNode = mapper.readTree(input);
+
+        if (isSQSEvent(rootNode)) {
```

Comment:
Given that is going to be held in memory in the form of a JsonNode I wouldn't bother then.
I'd do:

```
final byte[] bytes = inputStream.readAllBytes();
Optional<SQSEvent> sqsEvent = readSQSEvent(bytes);
// which is the following wrapper with exception handling
objectMapper.readValue(bytes, SQSEvent.class);

if (sqsEvent.isPresent()) { ... sqs ... }
// same thing for other types, easily you can handle API Gateway V1 too this way w/o getting into JSON schema details
```

Reply:
yeah, OK - it's using exceptions to control execution flow, but alright.  Agree it's more mem efficient.